core/src/main/java/org/apache/gravitino/storage/relational/service/CommonMetaService.java

```diff
@@ -37,22 +37,20 @@ public Long getParentEntityIdByNamespace(Namespace namespace) {
         !namespace.isEmpty() && namespace.levels().length <= 3,
         "Namespace should not be empty and length should be less than or equal to 3.");
     Long parentEntityId = null;
-    for (int level = 0; level < namespace.levels().length; level++) {
-      String name = namespace.level(level);
-      switch (level) {
-        case 0:
-          parentEntityId = MetalakeMetaService.getInstance().getMetalakeIdByName(name);
-          continue;
-        case 1:
-          parentEntityId =
-              CatalogMetaService.getInstance()
-                  .getCatalogIdByMetalakeIdAndName(parentEntityId, name);
-          continue;
-        case 2:
-          parentEntityId =
-              SchemaMetaService.getInstance().getSchemaIdByCatalogIdAndName(parentEntityId, name);
-          break;
-      }
+    if (namespace.levels().length > 0) {
```

Comment:
How about using `namespace.levels().length >=1` here? The same treatment is also applied below.

Reply:
parentEntityId only needs to be set once, if the level is 3 it will be set 3 times above.

Reply:
@justinmclean Hi, according to the current storage structure design, if the namespace has 3 levels, it is reasonable to recursively obtain the parentEntityId 3 times, because we cannot guarantee that the parent resource can be obtained uniquely through Name, so we need to use parentEntityId and Name to locate the upper-level resource at the same time.

Reply:
> @xloya
> 
> Thank you for your suggestion. I guess I missed something.
> 
> ```
> if (namespace.levels().length >= 1) {
>       parentEntityId = MetalakeMetaService.getInstance().getMetalakeIdByName(namespace.level(0));
> }
> 
> if (namespace.levels().length >= 2) {
>       parentEntityId =
>           CatalogMetaService.getInstance()
>               .getCatalogIdByMetalakeIdAndName(parentEntityId, namespace.level(1));
> }
> 
> if (namespace.levels().length >= 3) {
>       parentEntityId =
>           SchemaMetaService.getInstance()
>               .getSchemaIdByCatalogIdAndName(parentEntityId, namespace.level(2));
> }
> ```
> 
> Is it correct that you mentioned changing it like this? If it is, we will proceed with the change like this.

Yes, I think this can better express the Namespace levels:
```
if (namespace.levels().length >= 1) {
      ......
}

if (namespace.levels().length >= 2) {
      ......
}

if (namespace.levels().length == 3) {
      ......
}
```

Reply:
@xloya 
I think we'd better change the code if multiple people's opinions are set in one direction! Good evening.

Reply:
I still think there is no need to set parentEntityId more than once here and recursion is certainly not needed. Step though the code where level is 1,2 or 3 and you'll see it is doing work that is not required.

Reply:
For example if level is 3 then parentEntityId will be set three times with 2 of those values being thrown away. In this order:
1. via getMetalakeIdByName (result overwritten)
2. via getCatalogIdByMetalakeIdAndName  (result overwritten)
3. via getSchemaIdByCatalogIdAndName

Why not just obtain it by directly calling getSchemaIdByCatalogIdAndName?

e.g reverse the order of the if statements and change them to if/else ifs.

Reply:
I think what @khmgobe is reasonable, it is more readable than the original code, we can also improve like what @xloya mentioned.

The only bad thing is that `parentEntityId` is mutable, which is not a good programming practice, but I think it is acceptable as there's no side-effect here.

@khmgobe would you please modify the code like what @xloya suggested?

Reply:
@jerryshao @justinmclean 

Is there a consensus? 
Please let me know if you don't.

If it's matched, we're going to make a change.

The reason for asking again is that I think I'll change it
when it's decided for sure because I'm afraid there will be a change in the future.

Reply:
Yes, please go ahead.